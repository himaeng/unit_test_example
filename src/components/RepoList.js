import React, { Component, PropTypes } from 'react';
import {
  ListView,
  View,
  Text
} from 'react-native';

import RepoItem from './RepoItem';

const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id});

RepoList.propTypes = {
  repos: PropTypes.array.isRequired,
  selected: PropTypes.number,
  selectRepo: PropTypes.func
}

function RepoList (props) {
  const { repos, selected, selectRepo } = props;
  const dataSource = ds.cloneWithRows(repos);

  return (
    <ListView
      dataSource={dataSource}
      renderRow={(repo) => (
        <RepoItem
          isSelected={repo.id === selected}
          repo={repo}
          selectRepo={selectRepo}
        />
      )}
      contentContainerStyle={{ paddingVertical: 40 }}
    />
  );
};

export default RepoList;
