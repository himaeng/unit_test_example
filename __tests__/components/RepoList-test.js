import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import {
  ListView,
  View,
  Text
} from 'react-native';

import RepoList from '../../src/components/RepoList';
import { repos } from '../../config/jest/mockData';

describe('RepoList', () => {
  it('สามารถ render RepoList พร้อม mock data ได้อย่างถูกต้อง', () => {
    const component = renderer.create(
      <RepoList
        repos={repos}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('สามารถ render RepoList แบบ shallow อย่างถูกต้อง', () => {
    const ds = new ListView.DataSource(
      {
        rowHasChanged: (r1, r2) => (r1.id !== r2.id)
      });
    const dataSource = ds.cloneWithRows(repos);

    const wrapper = shallow(
      <RepoList
        repos={repos}
      />
    );

    expect(wrapper.contains(
      <ListView
        dataSource={dataSource}
        renderRow={(repo) => {
          <View>
            <Text>{repo.name}</Text>
            <Text>{`${repo.stargazers_count} stars`}</Text>
          </View>
        }}
      />
    ))
  });
});
