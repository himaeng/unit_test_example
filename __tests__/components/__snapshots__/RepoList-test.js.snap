// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList สามารถ render RepoList พร้อม mock data ได้อย่างถูกต้อง 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingVertical": 40,
    }
  }
  dataSource={
    ListViewDataSource {
      "items": 3,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Test 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        50000 stars
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Test 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        10000 stars
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Test 3
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        499 stars
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
